### PR TITLE
Tiny DPR controller

### PR DESCRIPTION
**Pull request by JacobMalling**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [x] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [x] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Core.ApplicationServices/GDPR/DataProcessingRegistrationApplicationService.cs [hunk 1]
```diff
@@ -503,5 +504,20 @@ public Result<(DataProcessingRegistration registration, IEnumerable<DataProcessi
                 return registration;
             });
         }
+
+        public IQueryable<DataProcessingRegistration> GetDataProcessingRegistrationsByOrganization(int orgId, params IDomainQuery<DataProcessingRegistration>[] conditions)
```
Comment (mrjsawdk) [inline Core.ApplicationServices/GDPR/DataProcessingRegistrationApplicationService.cs:508]: Organization behøves ikke at være påkrævet og hvis den gør, så skal du nok lige huske access control her :-)

Comment (mrjsawdk) [reply Core.ApplicationServices/GDPR/DataProcessingRegistrationApplicationService.cs:508]: Og hvis org id fjernes skal du stadig huske at filtrere base query ift. enten FULL cross org access eller alle brugerens orgs (se i IT-system usage service i min branch)

Comment (mrjsawdk) [reply Core.ApplicationServices/GDPR/DataProcessingRegistrationApplicationService.cs:508]: husk også en unit test

File: Core.ApplicationServices/GDPR/DataProcessingRegistrationUuidExtensionService.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        public DataProcessingRegistrationUuidExtensionService(IDataProcessingRegistrationApplicationService dataProcessingRegistrationApplicationService, IOrganizationService organizationService)
+        {
+            _dataProcessingRegistrationApplicationService = dataProcessingRegistrationApplicationService;
+            _organizationService = organizationService;
+        }
+
+        public Result<IQueryable<DataProcessingRegistration>, OperationError> GetDataProcessingRegistrationsByOrganization(Guid orgUuid, params IDomainQuery<DataProcessingRegistration>[] conditions)
+        {
+            var orgResult = _organizationService.GetOrganization(orgUuid);
```
Comment (mrjsawdk) [inline Core.ApplicationServices/GDPR/DataProcessingRegistrationUuidExtensionService.cs:23]: Du kan bruge Select metoden på result. Så slipper du for "IF" then return.. så er det bare 1 statement

File: Core.ApplicationServices/GDPR/IDataProcessingRegistrationApplicationService.cs [hunk 3]
```diff
@@ -55,5 +56,7 @@ public interface IDataProcessingRegistrationApplicationService
         Result<DataProcessingRegistrationOversightDate, OperationError> AssignOversightDate(int id, DateTime oversightDate, string oversightRemark);
         Result<DataProcessingRegistrationOversightDate, OperationError> ModifyOversightDate(int id, int oversightDateId, DateTime oversightDate, string oversightRemark);
         Result<DataProcessingRegistrationOversightDate, OperationError> RemoveOversightDate(int id, int oversightDateId);
+
+        IQueryable<DataProcessingRegistration> GetDataProcessingRegistrationsByOrganization(int orgId, params IDomainQuery<DataProcessingRegistration>[] conditions);
```
Comment (mrjsawdk) [inline Core.ApplicationServices/GDPR/IDataProcessingRegistrationApplicationService.cs:60]: Hvis du laver en Query(conditions) så behøves du ikke at have org id på

File: Core.ApplicationServices/GDPR/IDataProcessingRegistrationUuidExtensionService.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+using Core.DomainModel.Result;
+using Core.DomainServices.Queries;
+using System;
+using System.Linq;
+
+namespace Core.ApplicationServices.GDPR
+{
+    public interface IDataProcessingRegistrationUuidExtensionService
+    {
+        Result<IQueryable<DataProcessingRegistration>, OperationError> GetDataProcessingRegistrationsByOrganization(Guid orgUuid, params IDomainQuery<DataProcessingRegistration>[] conditions);
```
Comment (mrjsawdk) [inline Core.ApplicationServices/GDPR/IDataProcessingRegistrationUuidExtensionService.cs:11]: org id ikke påkrævet

Comment (mrjsawdk) [reply Core.ApplicationServices/GDPR/IDataProcessingRegistrationUuidExtensionService.cs:11]: og så er den her service vel heller ikke relevant længere

File: Tests.Unit.Presentation.Web/Tests.Unit.Presentation.Web.csproj [hunk 5]
```diff
@@ -205,6 +205,8 @@
     <Compile Include="Controllers\CronStringHelperTests.cs" />
     <Compile Include="DomainServices\Contract\QueryBySystemUsageUuidTest.cs" />
     <Compile Include="DomainServices\Contract\QueryBySystemUuidTest.cs" />
+    <Compile Include="DomainServices\DPR\QueryBySystemUsageUuidTest.cs" />
+    <Compile Include="DomainServices\DPR\QueryBySystemUuidTest.cs" />
```
Comment (mrjsawdk) [inline Tests.Unit.Presentation.Web/Tests.Unit.Presentation.Web.csproj:209]: hvor er API testen af den simple controller? Skal vi ikke have den med?

**Review by mrjsawdk — APPROVED** [review]
Der skal lige skrues lidt men ellers fint

File: Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/DataProcessingRegistrationV2Controller.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+        /// <summary>
+        /// Returns all Data-Processing-Registrations in the requested organization available to the user
+        /// </summary>
+        /// <param name="organizationUuid">Organization UUID filter</param>
+        /// <param name="systemUuid">System UUID filter</param>
+        /// <param name="systemUsageUuid">SystemUsage UUID filter</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("")]
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/DataProcessingRegistrationV2Controller.cs:41]: Mangler også en GET/{uuid}

**Review by mrjsawdk — APPROVED** [review]
mangler lige "fætter-index". Ellers 👍

**Comment by JacobMalling** [comment]
Så er den klar til merge

**Pull request merged**